### PR TITLE
Add docker configuration files from fork

### PR DESCRIPTION
**Pull request by chris-tbd**
This pull request primarily focuses on integrating Docker into the project. This includes the addition of Docker-related files, the setup of a Docker workflow on GitHub, and the modification of `package.json` to include Docker-related scripts. 

Docker Integration:

* [`.dockerignore`](diffhunk://#diff-2f754321d62f08ba8392b9b168b83e24ea2852bb5d815d63e767f6c3d23c6ac5R1-R5): Added a `.dockerignore` file to specify files and directories that Docker should ignore. This helps to avoid unnecessarily sending large or sensitive files and directories to the Docker daemon and potentially adding them to images.
* [`Dockerfile`](diffhunk://#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R1-R19): Added a Dockerfile to define how to build a Docker image for the application. This includes setting up the Node.js environment, copying the application files, installing dependencies, and setting up dbmate.
* [`docker-compose.yaml`](diffhunk://#diff-3fde9d1a396e140fefc7676e1bd237d67b6864552b6f45af1ebcc27bcd0bb6e9R1-R39): Added a `docker-compose.yaml` file to define and run multi-container Docker applications. This sets up two services: `postgresdb` and `pfi-app`.

GitHub Workflow:

* [`.github/workflows/docker-image-publish.yml`](diffhunk://#diff-be8608a0fb42a939ac410c1d809b2edc9e9208589da6deae1863450d6af512c3R1-R55): Added a GitHub workflow to automatically build and publish Docker images when changes are pushed to the `main` branch or when a pull request is made to the `main` branch.

Package.json Modifications:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L60-R66): Added several Docker-related scripts to `package.json`, including `docker:up`, `docker:down`, `docker:logs`, `docker:build`, and `docker:down-wipe`. These scripts provide convenient commands for managing Docker containers and images.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,6 +57,12 @@
     "lint:fix": "eslint . --ext .ts --fix",
     "seed-offerings": "npm run _start -- dist/seed-offerings.js",
     "server": "npm run _start -- dist/main.js",
-    "test": "npm run example-create-issuer && npm run seed-offerings && npm run example-create-customer && npm run example-issue-credential && npm run example-e2e-exchange"
+    "try": "npm run _debug -- dist/client-test.js",
```
Comment (michaelneale) [inline package.json:60]: can remove this as no longer used

**Review by michaelneale — CHANGES_REQUESTED** [review]
needs some docs on how to use it with docker/compose (probably in README) and one comment to remove the try command

File: package.json (hunk 1 above)
Comment (chris-tbd) [reply package.json]: @michaelneale removed

**Comment by chris-tbd** [comment]
@michaelneale right now `docker-compose` isn't bulletproof because it's not running the db migration scripts, I'm trying to unblock deployment and can manually run those scripts by logging into the cluster. Was talking with @leordev and we'll work on getting those migration scripts in, but until that point this isn't meant for just anyone to use quite yet

**Review by michaelneale — APPROVED** [review]

**Pull request merged**